src/include/common/null_mask.h

```diff
@@ -108,18 +113,32 @@ class NullMask {
 
     inline bool isNull(uint32_t pos) const { return isNull(data, pos); }
 
-    inline uint64_t* getData() { return data; }
+    // const because updates to the data must set mayContainNulls if any value
+    // becomes non-null
+    // Modifying the underlying data shuld be done with setNull or copyFromNullData
+    inline const uint64_t* getData() { return data; }
 
     static inline uint64_t getNumNullEntries(uint64_t numNullBits) {
         return (numNullBits >> NUM_BITS_PER_NULL_ENTRY_LOG2) +
                ((numNullBits - (numNullBits << NUM_BITS_PER_NULL_ENTRY_LOG2)) == 0 ? 0 : 1);
     }
 
-    // This function returns true if we have copied a nullBit with value 1 (indicate a null
-    // value) to dstNullEntries.
+    // @brief Copies bitpacked null flags from one buffer to another
```

Comment:
That style is used in the public interfaces. Is there a different style used internally?

Maybe it doesn't need every parameter documented, but I had to dig into the implementation to confirm that the offsets refer to bit offsets, not byte offsets (which I suspected, but byte offsets are used almost everywhere else).

Reply:
Yep, we usually only do that in public interfaces.
> Maybe it doesn't need every parameter documented, but I had to dig into the implementation to confirm that the offsets refer to bit offsets, not byte offsets (which I suspected, but byte offsets are used almost everywhere else).

I think you can state in the comment that these offsets refer to bit offsets instead of byte offsets.